### ISSUE TITLE
Implement autograd.

### ISSUE THREAD

**Issue by rohansingh9001** [RoboticsClubIITJ/ML-DL-implementation#64]
NOTE: Can only be worked on after issue #62 is completed and closed.

Autograd is a feature which lets us backpropagate gradients by automatically computing the gradients on a Tensor.

Such features come in very handy when implementing deep neural networks. 

You can use the following resource to implement autograd - https://medium.com/@a.nikishaev/making-backpropagation-autograd-mnist-classifier-from-scratch-in-python-bec4f05ce09f

**Comment by thisis-nkul** [comment]
I am willing to work on this. @rohansingh9001 please assign this to me if possible.
